src/blockchain/pop/vbk_block_tree.cpp

```diff
@@ -106,11 +107,19 @@ bool VbkBlockTree::addPayloads(const VbkBlock& block,
 }
 
 void VbkBlockTree::removePayloads(const VbkBlock& block,
-                                  std::vector<payloads_t>& payloads) {
+                                  const std::vector<payloads_t>& payloads) {
   auto* index = getBlockIndex(block.getHash());
-  assert(index);  // must exist
+  removePayloads(index, payloads);
```

Comment:
I think it should be possible to automatically detect payloads containing in the block.

Reply:
This method does not remove all payloads from the block, this method just remove payloads that are provided.